src/DotNetWorker.ApplicationInsights/FunctionsTelemetryModule.cs

```diff
@@ -20,18 +24,36 @@ public void Initialize(TelemetryConfiguration configuration)
 
             _listener = new ActivityListener
             {
-                ShouldListenTo = source => source.Name.StartsWith("Microsoft.Azure.Functions.Worker"),
+                ShouldListenTo = source => source.Name.StartsWith(TraceConstants.FunctionsActivitySource),
                 ActivityStarted = activity =>
                 {
                     var dependency = _telemetryClient.StartOperation<DependencyTelemetry>(activity);
-                    dependency.Telemetry.Type = "Azure.Functions";
-                    activity.SetCustomProperty("_depTel", dependency);
+                    activity.SetCustomProperty(DependencyTelemetryKey, dependency);
                 },
                 ActivityStopped = activity =>
                 {
-                    var dependency = activity.GetCustomProperty("_depTel") as IOperationHolder<DependencyTelemetry>;
-                    _telemetryClient.StopOperation(dependency);
-                    dependency?.Dispose();
+                    // Check for Exceptions events
+                    foreach (ActivityEvent activityEvent in activity.Events)
+                    {
+                        TrackExceptionTelemetryFromActivityEvent(activityEvent, _telemetryClient);
+                    }
+
+                    if (activity.GetCustomProperty(DependencyTelemetryKey) is IOperationHolder<DependencyTelemetry> dependencyHolder)
+                    {
+                        var dependency = dependencyHolder.Telemetry;
+
+                        foreach (var item in activity.Tags)
```

Comment:
nit: I think it's not needed and done within `StartOperation` https://github.com/microsoft/ApplicationInsights-dotnet/blob/main/BASE/src/Microsoft.ApplicationInsights/TelemetryClientExtensions.cs#L393

Reply:
This doesn't capture any tags added by someone later, though -- like inside the function itself with `Activity.Current.SetTag()`.

I had added a test for that -- is that not expected to work for auto-tracking stuff like this?

If I recall -- I copied the behavior from here: 
- https://github.com/microsoft/ApplicationInsights-dotnet/blob/main/WEB/Src/DependencyCollector/DependencyCollector/Implementation/EventHandlers/DiagnosticsEventHandlerBase.cs#L83

Which gets called from places like this:
- https://github.com/microsoft/ApplicationInsights-dotnet/blob/main/WEB/Src/DependencyCollector/DependencyCollector/Implementation/EventHandlers/ServiceBusDiagnosticsEventHandler.cs#L100

Reply:
yep, you're right, thanks for checking!